src/ShaderGen.Tests/TestUtil.cs

```diff
@@ -250,5 +262,217 @@ public static IReadOnlyCollection<(string fieldName, object aValue, object bValu
 
             return Unsafe.Read<T>(floats);
         }
+
+        /// <summary>
+        /// Gets a set of random floats.
+        /// </summary>
+        /// <param name="floatCount">The number of floats.</param>
+        /// <param name="minMantissa">The minimum mantissa.</param>
+        /// <param name="maxMantissa">The maximum mantissa.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentOutOfRangeException">minMantissa
+        /// or
+        /// maxMantissa</exception>
+        public static float[] GetRandomFloats(int floatCount, int minMantissa = -126, int maxMantissa = 128)
+        {
+            Random random = _randomGenerators.Value;
+            float[] floats = new float[floatCount];
+            for (int i = 0; i < floatCount; i++)
+            {
+                floats[i] = (float)((random.NextDouble() * 2.0 - 1.0) * Math.Pow(2.0, random.Next(minMantissa, maxMantissa)));
+                //floats[i] = (float)(random.NextDouble() * floatRange * 2f) - floatRange;
+            }
+
+            return floats;
+        }
+
+        #region ToMemorySize from https://github.com/webappsuk/CoreLibraries/blob/fbbebc99bc5c1f2e8b140c6c387e3ede4f89b40c/Utilities/UtilityExtensions.cs#L2951-L3116
+        private static readonly string[] _memoryUnitsLong =
+        {
+            " byte",
+            " kilobyte",
+            " megabyte",
+            " gigabyte",
+            " terabyte",
+            " petabyte",
+            " exabyte"
+        };
+
+        private static readonly string[] _memoryUnitsShort = { "B", "KB", "MB", "GB", "TB", "PB", "EB" };
+
+        /// <summary>
+        /// Converts a number of bytes to a friendly memory size.
+        /// </summary>
+        /// <param name="bytes">The bytes.</param>
+        /// <param name="longUnits">if set to <see langword="true" /> use long form unit names instead of symbols.</param>
+        /// <param name="decimalPlaces">The number of decimal places between 0 and 16 (ignored for bytes).</param>
+        /// <param name="breakPoint">The break point between 0 and 1024 (or 0D to base on decimal points).</param>
+        /// <returns>System.String.</returns>
+        public static string ToMemorySize(
```

Comment:
Do we actually need all of these overloads? If not, let's just include the one(s) we actually use.

Reply:
Sorry, it was a cut'n'paste from my other project, I figured I'd put them in here in case they were useful in future.  I'll remove the unused overloads.